server/mscalendar/availability.go

```diff
@@ -106,6 +112,37 @@ func (m *mscalendar) setUserStatuses(users store.UserIndex, schedules []*remote.
 	return utils.JSONBlock(schedules), nil
 }
 
+func (m *mscalendar) notifyUpcomingEvent(mattermostUserID string, items []remote.ScheduleItem) {
+	var timezone string
+	for _, scheduleItem := range items {
```

Comment:
I was going to suggest adjusting this function to take one `remote.ScheduleItem`, and have the caller loop through the different items (and pass in the timezone). But the strategy here with lazy loading the timezone seems worth it, considering how often this job runs.